src/tests/ftest/nvme/nvme_pool_extend.py

```diff
@@ -0,0 +1,251 @@
+            thrd.join()
+
+    def run_nvme_pool_extend(self, num_pool):
+        """Run Pool Extend
+            Args:
+             int : total pools to create for testing purposes.
+        """
+        # Create a pool
+        pool = {}
+        pool_uuid = []
+        total_servers = len(self.hostlist_servers)
+
+        # Extend one of the ranks (or server)
+        # rank index starts from zero
+        rank = total_servers
+
+        for val in range(0, num_pool):
+            pool[val] = TestPool(self.context, dmg_command=self.dmg_command)
+            pool[val].get_params(self)
+            # Split total SCM and NVME size for creating multiple pools.
+            pool[val].scm_size.value = int(pool[val].scm_size.value /
+                                           num_pool)
+            pool[val].nvme_size.value = int(pool[val].nvme_size.value /
+                                            num_pool)
+            pool[val].create()
+            pool_uuid.append(pool[val].uuid)
+
+        for val in range(0, num_pool):
+            self.pool = pool[val]
+            for oclass, api, test in product(self.ior_dfs_oclass,
+                                             self.ior_apis,
+                                             self.ior_test_sequence):
+                self.run_ior_thread("Write", oclass, api, test)
+
+                scm_size = self.pool.scm_size
+                nvme_size = self.pool.nvme_size
+                self.pool.display_pool_daos_space("Pool space: Beginning")
+                pver_begin = self.get_pool_version()
+
+        # Start the additional servers and extend the pool
+        self.log.info("Extra Servers = %s", self.extra_servers)
+        self.start_additional_servers(self.extra_servers)
+        # Give sometime for the additional server to come up.
+        time.sleep(5)
+        self.log.info("Pool Version at the beginning %s", pver_begin)
+        output = self.dmg_command.pool_extend(self.pool.uuid,
+                                              rank, scm_size,
+                                              nvme_size)
+        self.log.info(output)
+        fail_count = 0
+        while fail_count <= 20:
+            pver_extend = self.get_pool_version()
+            time.sleep(15)
+            fail_count += 1
+            if pver_extend > pver_begin:
+                break
+
+        self.log.info("Pool Version after extend %s", pver_extend)
+        # Check pool version incremented after pool exclude
+        self.assertTrue(pver_extend > pver_begin,
+                        "Pool Version Error:  After extend")
+
+        for val in range(0, num_pool):
+            self.pool = pool[val]
+            for oclass, api, test in product(self.ior_dfs_oclass,
+                                             self.ior_apis,
+                                             self.ior_test_sequence):
+                # Verify the data after pool extend
+                self.run_ior_thread("Read", oclass, api, test)
```

Comment:
This is not going to work because container uuid is getting created for each call (line 113) so when you will try to read you need to retain the original container uuid stored somewhere and use that during read

Reply:
Oh, I see. Will do. Thanks.

Reply:
Dome.